pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -241,26 +241,24 @@ func (esc *endpointSliceController) updateEndpointCacheForSlice(hostName host.Na
 		// Draining tracking is only enabled if persistent sessions is enabled.
 		// If we start using them for other features, this can be adjusted.
 		healthStatus := endpointHealthStatus(svc, e)
-		for _, a := range e.Addresses {
-			pod, expectedPod := getPod(esc.c, a, &metav1.ObjectMeta{Name: slice.Name, Namespace: slice.Namespace}, e.TargetRef, hostName)
-			if pod == nil && expectedPod {
-				continue
+		pod, expectedPod := getPod(esc.c, e.Addresses[0], &metav1.ObjectMeta{Name: slice.Name, Namespace: slice.Namespace}, e.TargetRef, hostName)
```

Comment:
same comment as in endpoint_builder.go

Reply:
Hi @howardjohn, yeah, I change this because we do not hope to create duplicated `IstioEndpoint` here, but it seems that it's incorrect. I added a `podMap` to eliminate duplicated `IstioEndpoint` instance now, please check the new submit. Thanks!

Reply:
Hi @howardjohn, thanks to point this out, I change this because I want to remove the IstioEndpoint duplication for pod, however I revert the `for _, a := range e.Addresses {` loop because `endpointslice` collects the IP addresses based on address type, so there should be no duplicated endpoints in `e.Addresses`, they always belong to different endpoints.

Reply:
Yes I understand the change in the loop. But the `getPod` logic is broken now.

Reply:
Hi @howardjohn, I give more explanation for `getPod` in above comment. We can discuss it there.

Reply:
Hi @howardjohn, there are some changes for `istioEndpoint` building from `L#276~288` in this file, this change eliminate the duplicated istioEndpoint if a pod has multiple addresses, and all the changes are protected by feature flag.